Added PHP 8.0 support

gennadiylitvinyuk opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | yes
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

Added PHP 8.0 support

- Added PHP 8.0 constraint to composer.json
- Removed PHP less 7.3 support from composer.json
- Modified composer.json to implement phpunit 9.3
- Fixed phpunit tests and config to be compatible with the phpunit 9.3
- Added PHP 8.0 to .travis.yml
- Removed PHP less 7.3 support from .travis.yml

Signed-off-by: Gennadiy Litvinyuk <gennadiy@gmail.com>

gennadiylitvinyuk committed:
Added PHP 8.0 support

- Added PHP 8.0 constraint to composer.json
- Removed PHP less 7.3 support from composer.json
- Modified composer.json to implement phpunit 9.3
- Fixed phpunit tests and config to be compatible with the phpunit 9.3
- Added PHP 8.0 to .travis.yml
- Removed PHP less 7.3 support from .travis.yml

Added PHP 8.0 support

- Added PHP 8.0 constraint to composer.json
- Removed PHP less 7.3 support from composer.json
- Modified composer.json to implement phpunit 9.3
- Fixed phpunit tests and config to be compatible with the phpunit 9.3
- Added PHP 8.0 to .travis.yml
- Removed PHP less 7.3 support from .travis.yml

Signed-off-by: Gennadiy Litvinyuk <gennadiy@gmail.com>

gennadiylitvinyuk commented:
Implementation of #12

froschdesign commented on phpunit.xml.dist:
```diff
@@ -1,17 +1,17 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
```
Please revert this line. The local schema definition should be used, otherwise we have to control the version in the Composer configuration file and here.

froschdesign commented:
Travis CI is gone and the [new workflow for continuous integration](https://github.com/laminas/laminas-continuous-integration-action) must be added that the tests can be run. (See also on other components for help, e.g. [laminas-feed](https://github.com/laminas/laminas-feed/pull/35).)

gennadiylitvinyuk committed:
CI Configuration

CI Configuration: from Travis to Github Actions
Reverted phpunit schema
Bumped up laminas-coding-standard
CS Fixes

Signed-off-by: Gennadiy Litvinyuk <gennadiy@gmail.com>

gennadiylitvinyuk committed:
CS Fix

Signed-off-by: Gennadiy Litvinyuk <gennadiy@gmail.com>

weierophinney commented on src/ComposerJsonRetrievalTrait.php:
```diff
@@ -11,11 +11,11 @@
 use Composer\Factory;
 use Composer\Json\JsonFile;
 
+use function call_user_func;
```
Since this patch updates the skeleton to set the minimum PHP version to 7.3, we can stop using `call_user_func()`, and instead just call the callable directly (as 7.2 forwards allows any callable to be directly invoked). Please remove this import, and then update the `getComposerJson()` call such that:

```php
$composerFile = call_user_fun($this->composerFileFactory);
```

becomes:

```php
$composerFile = ($this->composerFileFactory)();
```

gennadiylitvinyuk replied:
Done

weierophinney commented on src/OptionalPackagesInstaller.php:
```diff
@@ -15,6 +15,11 @@
 use Composer\Package\Version\VersionParser;
 use Composer\Plugin\PluginInterface;
+
+use function call_user_func;
```
Same comment as for the `ComposerJsonRetrievalTrait`, only this time, the function to update will be `runInstaller()`.

gennadiylitvinyuk replied:
Done

gennadiylitvinyuk committed:
Replaced call_user_func() with callables

Signed-off-by: Gennadiy Litvinyuk <gennadiy@gmail.com>

samsonasik commented:
Thank you @gennadiylitvinyuk
